### PR TITLE
Go: update .meta/cofing.json files and add exercises to config.json file

### PR DESCRIPTION
**Pull request by oanaOM**
Please review and let me know if I need to add/remove anything else to .meta/config.json file for each exercise(#1194 ) and the general config.json with all the exercises (#1195 )

### CODE REVIEW

File: languages/go/config.json [hunk 1]
```diff
@@ -9,7 +9,26 @@
   },
   "ignore_pattern": "example(?!.*test)",
   "exercises": {
-    "concept": [],
+    "concept": [
+      {
+        "slug": "basic-strings",
```
Comment (ErikSchierboom) [inline languages/go/config.json:14]: The slug doesn't currently match the concept of the exercise (`strings`). This is not a requirement, but it _is_ common to have the two align. You could thus consider renaming the slug to:

```suggestion
        "slug": "strings",
```

Do note that this means that the `exercises/concept/basic-strings` directory also has to be renamed to `strings`.

The same goes for the `basic-slices` slug below. If you'd like to do this, I would suggest doing it in a follow-up PR as to not make this PR too big.

**Comment by oanaOM** [comment]
@ErikSchierboom I've renamed the directory for the 2 concept exercises and also updated the go/exercises/concept/README.md file to point to those exercises. 

Just after I've committed my changes, I've realized that they will go under the same PR, sorry about this :(

Also, I've noticed that the check for dead links has failed. Is this of changes?

**Comment by oanaOM** [comment]
I've just fixed the dead links.

**Comment by ErikSchierboom** [comment]
@SleeplessByte We don't yet have automatic JSON formatting. We could add it, but we only support Markdown and C# formatting at the moment.

File: languages/go/exercises/concept/numbers/.meta/config.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+    "authors": [
```
Comment (SleeplessByte) [inline languages/go/exercises/concept/numbers/.meta/config.json:2]: Can you normalise this using 2 spaces?

File: languages/go/exercises/concept/slices/.meta/config.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
-	"contributors": [
+	"authors": [
```
Comment (SleeplessByte) [inline languages/go/exercises/concept/slices/.meta/config.json:2]: Can you normalise this using 2 spaces?

File: languages/go/exercises/concept/strings/.meta/config.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+	"authors": [
```
Comment (SleeplessByte) [inline languages/go/exercises/concept/strings/.meta/config.json:2]: Can you normalise this using 2 spaces?

File: languages/go/exercises/concept/numbers/.meta/config.json (hunk 2 above)
Comment (oanaOM) [reply languages/go/exercises/concept/numbers/.meta/config.json]: Sure, done.

File: languages/go/exercises/concept/slices/.meta/config.json (hunk 3 above)
Comment (oanaOM) [reply languages/go/exercises/concept/slices/.meta/config.json]: Sure, done.

**Comment by oanaOM** [comment]
Great! I've formatted the JSON files and push this through .

**Comment by tehsphinx** [comment]
@oanaOM: Awesome!
@SleeplessByte: Can you have a look and unlock the PR?

**Comment by oanaOM** [comment]
Is this something I need to review?

**Comment by SleeplessByte** [comment]
@oanaOM If you can resolve the merge conflict on `languages/go/exercises/concept/basic-slices/.meta/config.json`, that'd be great. Easiest to reproduce that conflict locally is to update your local `master` branch, then checkout `go-config-files`, and merge in `master`.

image

**Comment by oanaOM** [comment]
@SleeplessByte I think I've solved the conflict. Could you check now?
I might need to add back that file `languages/go/exercises/concept/basic-slices/.meta/config.jso`  though

**Comment by SleeplessByte** [comment]
@oanaOM it renamed the file to `slices`/.meta/config.json`, which is correct if the exercise slug is `slices`.

A different thing I've noticed (cc @ErikSchierboom  correct me if I'm wrong), is that the **directories** are now correct, but the _file names_ are not.

`basic_slices.go` should be called `slices.go` etc. The same goes for all the other renames (`basic_strings.go` should be `strings.go`).

I'd suggest you fix the slices one, then we get this merged, and then someone fixes them across the track.

**Comment by ErikSchierboom** [comment]
> A different thing I've noticed (cc @ErikSchierboom correct me if I'm wrong), is that the directories are now correct, but the file names are not.

Good catch! While in principle the files _can_ be named differently, most test runners will expect them to have the same name as the slug/directory (possibly in a different casing). So your comment is indeed correct.

**Comment by SleeplessByte** [comment]
Yeah. The track maintenance tool, and most CI (track ci) also expects them to be like that.

@oanaOM as far as I'm concerned, if you can rename `basic_slices.go` and the test file to `slices.go` , this can be squash-merged ❤️

**Comment by oanaOM** [comment]
@SleeplessByte great eagle eye! I've renamed the files for both slices and strings concept exercises. Please review now.

**Comment by ErikSchierboom** [comment]
@SleeplessByte I've taken the liberty to merge this, as your suggestions were applied.

**Comment by tehsphinx** [comment]
@oanaOM: thanks a lot for you help!

**Pull request merged**